File: python/1st_week/MariMencoba/Tabel.py

class Tabel:
    # class untuk tabel

    def __init__(self, baris, kolom):
        self.baris = baris
        self.kolom = kolom

    # set baris
    def setBaris(self, b, k):
        self.kolom = k
        self.baris = b 
    
    # membuar baris
    def buatBaris(self, add, isi = []):
        # membuat garis tabel atas
        i = 0 
        for i in self.isi.len:
            j = 0
            for j in self.i.len() + self.add:
                print("-")
        print("\n")

        # membuat isi
        i, j = 0
        for i in self.isi.len:
            print("| ")
            print(i)
            for j in add-3:
                print(" ")
            print("|")
        print("\n")

        i = 0 
        for i in self.isi.len:
            j = 0
            for j in self.i.len() + self.add:
                print("-")
        print("\n")
    

File: cpp/1st_Week/MariMencoba/Tabel.cpp
#include <iostream>
#include <string>

using namespace std;

class Tabel
{
    private:
        int baris;
        int kolom;
    public:
    Tabel(){
    }

    Tabel(int b, int k){
        baris = b;
        kolom = k;
    }
    
    //method membuat baris
    void buatBaris(string isi[], int add){
        
        int i = 0, j = 0;

        //membuat garis tabel bagian atas
        for ( i = 0; i < add; i++)
        {
            for (j = 0; j < (isi[i].length() + add); j++)
            {
                cout << "-";
            }
        }
        cout << "" << endl;

        //menampilkan isi array
        for ( i = 0; i < add; i++)
        {
            cout << "| ";
            cout << isi[i];
            for (j = 0; j < add - 3; j++)
            {
                cout << " " << isi[i];
            }
            cout << "|";
        }
        cout << "" << endl;
        

        //membuat garis tabel bagian bawah
        for ( i = 0; i < add; i++)
        {
            for (j = 0; j < (isi[i].length() + add); j++)
            {
                cout << "-";
            }
        }
        cout << "" << endl;
    }

    ~Tabel(){
    }
};
